R/pkg/R/DataFrame.R

```diff
@@ -392,7 +392,11 @@ setMethod("coltypes",
                 }
 
                 if (is.null(type)) {
-                  stop(paste("Unsupported data type: ", x))
+                  specialtype <- specialtypeshandle(x)
```

Comment:
as line 383
`type <- PRIMITIVE_TYPES[[x]]`
here in line 399
`type <- specialtype`

numeric is one of the value from PRIMITIVE_TYPE, double is not